docs/rfc/rfc-1.adoc

```diff
@@ -435,13 +435,49 @@ As time passes and operators drop out of the system, a wallet becomes at risk
 of being able to meet the 51-of-100 threshold to produce signatures.
 Additionally, we want to avoid situations where operators are the custodians of
 a wallet for extended periods. To avoid these issues, once a wallet is older
-than the `wallet_max_age`, or if it drops below the liveness threshold (say, below 70 on
-a <<heartbeat,heartbeat>>), we motion to transfer the funds to another randomly
-selected wallet.
+than the `wallet_max_age`, or if it drops below the liveness threshold (say,
+below 70 on a <<heartbeat,heartbeat>>), we motion to
+<<closing-a-wallet,transfer the funds>> to another randomly selected wallet.
 
 Once a wallet no longer has funds and is not the primary wallet for new
-deposits, it can be closed and operators are no longer required to maintain
-it.
+deposits, it can be <<closing-a-wallet,closed>> and operators are no longer
+required to maintain it.
+
+[[closing-a-wallet]]
+==== Closing A Wallet
+Governable Parameters:
+
+- `wallet_min_btc`: The smallest amount of BTC a wallet can hold before we
+  attempt to close the wallet and transfer the funds to a randomly selected
+  wallet.
+
+When a wallet fails a <<heartbeat,heartbeat>> or ends a
+<<redemption,redemption>> with less than `wallet_min_btc` remaining, the
+operators begin the process of closing it. If there are still funds remaining,
+the <<operator-only,first operator>> posts a <<operator-only,reimbursable>>
+transaction to the ethereum chain declaring an intention to close the wallet.
+
+That operator proposes a BTC fee (as in <<sweeping,sweeping>>) and an ethereum
+block height commitment, and they use that to construct a P2PKH transaction
+moving the wallet's main UTXO to another randomishly chosen wallet where the
+source of randomness comes from the ethereum block hash at the pre-commited
+block height.
+
+After this transaction is complete, <<public-knowledge,the public>> is able
+to submit a SPV proof to let ethereum know that the funds were transferred
+nonfraudulently, and which account balances need to be updated (like the
+treasur which pays the bitcoin mining fee).
+
+*Notes*: We trust the honest majority to pick a new random wallet. Transferring
+the BTC to any address other than the P2PKH of one of the other wallet
+addresses is fraud, and is <<Punishment,punishable>>. Moving funds without
+sending a follow-up proof is <<Punishment,punishable>>. Failure to close a
```

Comment:
So the follow-up proof can be sent by anyone and if it's not sent (in what time?), the operators of the emptying wallet may be punished. So usually it will be operators of the wallet who send that proof (it's in their interest to do that), but theoretically this could be done by everybody, because everybody has a sufficient knowledge to do that. Do I understand correctly?

Reply:
e4717caa5fa2862dce4f2e184f670b441c6fc43b fleshes out the timeout mechanism here. We post a transaction when we notice that a wallet needs to be closed (anyone including operators can do it), and we use the hash of that tx to decide which wallet to send funds to. The timeout starts counting from when that tx clears.

Other than that, the rest of your understanding is spot on!